Server/ReasnAPI/ReasnAPI/Services/TagService.cs

```diff
@@ -1,16 +1,118 @@
 ﻿using ReasnAPI.Models.Database;
+using ReasnAPI.Models.DTOs;
+using System.Linq.Expressions;
+using System.Transactions;
+using Microsoft.EntityFrameworkCore;
+using ReasnAPI.Exceptions;
+using ReasnAPI.Mappers;
 
-namespace ReasnAPI.Services {
-    public class TagService (ReasnContext context) {
-        private readonly ReasnContext _context = context;
-
-        /* TODO: Create following functions for this class
-         * create
-         * update
-         * delete
-         * get by ID
-         * get list by filter
-         * get all
-         */
+namespace ReasnAPI.Services;
+public class TagService (ReasnContext context)
+{
+    public TagDto CreateTag(TagDto tagDto)
+    {
+        var tag = context.Tags.FirstOrDefault(r => r.Name == tagDto.Name);
+        if (tag is not null)
+        {
+            throw new BadRequestException("Tag already exists");
+        }
+
+        context.Tags.Add(tagDto.ToEntity());
+        context.SaveChanges();
+        return tagDto;
+    }
+
+    public TagDto UpdateTag(int tagId, TagDto tagDto, int eventId)
```

Comment:
You can create another UpdateTag method only to update tag, not for event, which will be for admin in order to update if it's missspelled or so like you have in Parameter. And why don't you have same logic for parameters? That's what I was saying - your approach look a lot different in services even tho it should be the same, it's misleading then, because you assume it will do what it says and in each service it does it differently. I think adding UpdateParameterForEvent would be great and change this one to UpdateTagForEvent, sounds right?

Reply:
i created one and renamed old one